lake/top/tech_maps.py

```diff
@@ -108,13 +111,62 @@ def GF_Tech_Map(depth, width) -> dict:
         }
     }
 
-    ports.append(single_port)
+    dual_port_p0rw = {
+        # port RW
+        'clk': 'CLK_A',
+        'cen': 'CEN_A',
+        'write_enable': 'RDWEN_A',
+        'addr': 'A_A',
+        'data_in': 'D_A',
+        'data_out': 'Q_A',
+        'alt_sigs': {
+            # value, width
+            'T_Q_RST_A': (0, 1),
+            'T_LOGIC': (0, 1),
+            'MA_SAWL1': (0, 1),
+            'MA_SAWL0': (0, 1),
+            'MA_WL1': (0, 1),
+            'MA_WL0': (0, 1),
+            'MA_WRAS1': (0, 1),
+            'MA_WRAS0': (0, 1),
+            'MA_VD1': (0, 1),
+            'MA_VD0': (0, 1),
+            'MA_WRT': (0, 1),
+        }
+    }
 
-    tech_map = {
-        'name': f"IN12LP_S1DB_W{depth:05}B{width:03}M04S2_H",
-        'ports': ports,
-        'depth': depth,
-        'width': width
+    dual_port_p1r = {
+        # port RW
+        'clk': 'CLK_B',
+        'cen': 'CEN_B',
+        'read_addr': 'A_B',
+        'data_out': 'Q_B',
+        'alt_sigs': {
+            # value, width
+            'T_Q_RST_B': (0, 1),
+            'RDWEN_B': (1, 1),
+            'D_B': (0, width),
+        }
     }
 
+    if dual_port:
+        ports.append(dual_port_p0rw)
+        ports.append(dual_port_p1r)
+
+        tech_map = {
+            'name': f"IN12LP_SDPB_W{depth:05}B{width:03}M08S2_H",
+            'ports': ports,
+            'depth': depth,
+            'width': width
+        }
+    else:
+        ports.append(single_port)
+
+        tech_map = {
+            'name': f"IN12LP_S1DB_W{depth:05}B{width:03}M04S2_H",
+            'ports': ports,
+            'depth': depth,
+            'width': width
+        }
```

Comment:
there's some redundant code here - can set name and add the ports in the if/else then set the final tech map after

Reply:
Thanks for the review!